src/ngraph/runtime/gpu/gpu_emitter.hpp

```diff
@@ -87,18 +87,24 @@ namespace ngraph
                         dtypes.push_back(out[0].get_type());
                         auto ew_index =
                             cuda_emitter->build_elementwise<T>(dtypes, out[0].get_shape());
-                        writer << "gpu::invoke_primitive(ctx, " << ew_index << ", ";
-                        writer << "std::vector<void*>{" << args.front().get_name();
-                        for (size_t i = 1; i < args.size(); i++)
-                        {
-                            writer << ", " << args[i].get_name();
-                        }
-                        writer << "}.data(), ";
-                        writer << "std::vector<void*>{" << out[0].get_name() << "}.data()";
-                        writer << ");\n";
+                        writer << "void* input[] = {" << node_names(args) << "};\n";
+                        writer << "void* output[] = {" << node_names(out) << "};\n";
+                        writer << "gpu::invoke_primitive(ctx, " << ew_index
+                               << ", input, output);\n";
                     }
                     writer.block_end();
                 }
+
+            private:
+                static std::string node_names(const std::vector<GPU_TensorViewWrapper>& args)
```

Comment:
Being able to use node_names when the order is different would be nice.

Can we add an additional argument that specifies the order of the args? It should default to the order of args in the vector of tensor views, but some of our ops rearrange the tensor order in the invocation. 

e.g.
```
writer << "void* input[] = {" << node_names(args, {2,0,1}) << "};\n";

```

Reply:
Do you also want to skip certain values? like `{2, -1, 1}` would only yield `{arg[2], arg[1]}`

Reply:
or `{2, 1}'

Reply:
I think this is good, but I don't think it's necessary. It's easy to handle in the function you pass.

Reply:
I see what you mean. looks like easier to handle in gnu_emitter in this case.